### ISSUE TITLE
Enable control of use of caching in tf.keras.preprocessing.image_dataset_from_directory

### ISSUE THREAD

**Issue by kechan** [tensorflow/tensorflow#44354]
**System information**
- TensorFlow version (you are using): TF 2.3.0 
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
This concerns the API tf.keras.preprocessing.image_dataset_from_directory(...) I read through the doc as well as the source code, it doesn't seem like there's any control over behavior concerning cache() for the tf.data.Dataset. As best I could understand this, it won't be right to apply .cache(...) to the output tf dataset of this API since shuffling is done (esp. for train) and caching anything after, it freezes the initial shuffling order, such that it stays fixed at subsequent epochs. According to general wisdom, this is highly likely not desirable. The caching should be done before shuffling, but there's no way to specify this at API call time. The only workaround I could think of is to copy/paste the exact code, and write your own version.

**Will this change the current api? How?**
One can add this feature without affecting old behavior. Cache is simply assumed False unless supplied otherwise, and if it is false, the old code path should execute.

**Who will benefit with this feature?**
image_dataset_from_directory(...) is a nice convenient method and caching may improve performance by not repeating the image resizing processing. It can also help if your original directories are remote and may suffer outage or unpredictable lag.

**Any Other info.**

**Comment by sachinprasadhs** [comment]
tf.keras.preprocessing.image_dataset_from_directory will be deprecated from Tensorflow 2.9 version, prefer loading data with [tf.keras.utils.image_dataset_from_directory](https://www.tensorflow.org/api_docs/python/tf/keras/utils/image_dataset_from_directory), and then transforming the output [tf.data.Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset) with preprocessing layers.
If you still feel this feature needs to be added, feel free to open a new feature request. Thanks!
